OpenRA.Mods.Common/Traits/Player/ClassicProductionQueue.cs

```diff
@@ -52,16 +52,23 @@ protected override void Tick(Actor self)
 		{
 			// PERF: Avoid LINQ.
 			isActive = false;
+			var isActiveUnpaused = false;
```

Comment:
I don't understand this.

Reply:
I couldn't come up with anything better that isn't overly long.

Basically, this checks whether the queue is neither disabled nor paused.

Reply:
Why can't we track that using `isActive`?

Reply:
Wouldn't that make IsTraitDisabled and IsTraitPaused have the same effect?

I'm wary of that since isActive is used in other places.

Besides, I think the most correct behavior would be for the queue to be cleared when all Productions are disabled, but just not tick when some are enabled but all of those enabled are paused. That's not possible if we use isActive alone.

Reply:
In that case IMO the real problem is with the `isActive` name.  `isActive` should be used for pausing, while `isEnabled` or similar should be used for disabled/clearing production.